src/component/auth/SignUp.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+const SignUp = () => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const history = useHistory();
+
+  const handleRegistration = async () => {
+    const userData = { name, email, password };
+
+    try {
+      // Perform client-side validation if needed
+
+      // Send registration data to server
+      const response = await fetch("api/register", {
+        method: 'POST',
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json"
+        },
+        body: JSON.stringify(userData)
+      });
+
+      if (!response.ok) {
+        throw new Error('Registration failed');
+      }
+
+      // Registration successful, redirect to login page
+      history.push('/Login');
+
+      // Optionally, handle response data
+      const result = await response.json();
+      localStorage.setItem("user-info", JSON.stringify(result));
+    } catch (error) {
+      console.error('Error during registration:', error.message);
+      // Optionally, display an error message to the user
+    }
+  };
```

Comment:
Same in the signup also please update the code to use `useNavigate` instead of `useHistory`.
`useHistory` is not available in `react-router-dom` v6.x and onwards. Please ensure to test thoroughly after making this change.